ICAFIX/hcp_fix_multi_run

```diff
@@ -859,13 +866,24 @@ for fmri in $fmris ; do
 	for readme_fmri in ${fmris} ; do
 		echo "  ${readme_fmri}" >> ${readme_for_cifti_out}
 	done
-		
+
 	volume_out=${fmriNoExt}_hp${hp}_clean.nii.gz
 	${Caret7_Command} -volume-merge ${volume_out} -volume ${ConcatFolder}/${concatfmrihp}_clean.nii.gz -subvolume ${Start} -up-to ${Stop}
 	fslmaths ${volume_out} -div ${ConcatFolder}/${concatfmrihp}_vn -mul ${fmriNoExt}_hp${hp}_vn -add ${fmriNoExt}_mean ${volume_out}
+
+  if [ -f ${ConcatFolder}/${concatfmrihp}.ica/mc/prefiltered_func_data_mcf_conf_hp_clean.nii.gz ] ; then
+    ${Caret7_Command} -volume-merge ${fmriNoExt}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf_hp_clean.nii.gz -volume ${ConcatFolder}/${concatfmrihp}.ica/mc/prefiltered_func_data_mcf_conf_hp_clean.nii.gz -subvolume ${Start} -up-to ${Stop}
+    fslmeants -i ${fmriNoExt}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf_hp_clean.nii.gz -m ${ConcatFolder}/${concatfmrihp}.ica/mc/prefiltered_func_data_mcf_conf_hp_clean_mask.nii.gz -o $(dirname $fmri)/Movement_Regressors_hp${hp}_clean.txt --showall
+    cat $(dirname $fmri)/Movement_Regressors_hp${hp}_clean.txt | tail -$(fslval ${fmriNoExt}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf_hp_clean.nii.gz dim4) > $(dirname $fmri)/Movement_Regressors_hp${hp}_clean.txt
+  fi
+
 	Start=`echo "${Start} + ${NumTPS}" | bc -l`
 done
 
+if [ -f ${ConcatFolder}/${concatfmrihp}.ica/mc/prefiltered_func_data_mcf_conf_hp_clean.nii.gz ] ; then
```

Comment:
Is there a case where the mask file exists, but the data doesn't?  If you just want to make sure the mask file isn't kept around, just use `rm -f` on the mask, no conditional.

Reply:
The code as implemented will work correctly, I don't want to generate errors by trying to remove a non existent mask.

Reply:
`rm -f` will never give an error that a filename doesn't exist:

```
tim@timsdev:~$ rm -f
tim@timsdev:~$ rm -f kahgsdkfjghakdjfgasf
tim@timsdev:~$ 
```

Checking for the existence of a file other than the one you are removing is confusing, and you haven't clarified if that particular logic actually needs to be that way.  Do you ever want a file that has the mask name to be kept past that point in the script?